### ISSUE TITLE
gcc 7 seems cannot compile this proj?

### ISSUE THREAD

**Issue by courao** [SeanNaren/warp-ctc#71]
gcc 7 seems cannot compile this proj?

**Comment by ikuinen** [comment]
i also encounter this problem

**Comment by slavaGanzin** [comment]
No. Only gcc6. In case you have two or more gcc versions, this will work:

```
cmake -D CMAKE_C_COMPILER:string=gcc-6  -D CMAKE_CXX_COMPILER=c++-6  .. 
```

**Comment by cbasavaraj** [comment]
@slavaGanzin Thanks, useful for me just the day after you posted! Small typo though I think: g++-6 instead of c++-6. So:
`cmake -D CMAKE_C_COMPILER:string=gcc-6  -D CMAKE_CXX_COMPILER=g++-6  ../`

**Comment by cbasavaraj** [comment]
Addendum: I was getting errors during the make step
`CMake Error at test_gpu_generated_test_gpu.cu.o.cmake:279`
Tried with gcc-5, luckily this worked!
Whew, the amount of work to get this working!

**Comment by Honghe** [comment]
gcc-5 works on Ubuntu 18.04, too.
